### PR TITLE
Ovirt Scenario and User guidelines

### PR DESCRIPTION
**Pull request by arendej**
##### SUMMARY

follow-up to funixz:patch-8, setting up ovirt/rhv to have presence in scenario guides.

##### ISSUE TYPE

- Docs Pull Request

##### COMPONENT NAME

ovirt scenario guides in docs

##### ADDITIONAL INFORMATION

n/a

### CODE REVIEW

File: docs/docsite/rst/scenario_guides/ovirt_scenarios/faq.rst [hunk 1]
```diff
@@ -0,0 +1,34 @@
+.. _oVirt_faq:
+
+***************
+Ansible oVirt FAQ
```
Comment (Akasurde) [inline docs/docsite/rst/scenario_guides/ovirt_scenarios/faq.rst:9]:
```suggestion
*****************
Ansible oVirt FAQ
*****************


```

File: docs/docsite/rst/scenario_guides/ovirt_scenarios/faq.rst [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+***************
+
+ovirt_* modules
+===============
+
+What is oVirt? What is oVirt?
+---------------------------
+
+oVirt is the upsteam open-source edition of Red Hat Virtualization.
```
Comment (Akasurde) [inline docs/docsite/rst/scenario_guides/ovirt_scenarios/faq.rst:21]:
```suggestion
oVirt is the upstream open-source edition of Red Hat Virtualization. 
```

Comment (Akasurde) [inline docs/docsite/rst/scenario_guides/ovirt_scenarios/faq.rst:21]:
```suggestion
oVirt is the upstream open-source edition of Red Hat Virtualization. 
```

File: docs/docsite/rst/scenario_guides/ovirt_scenarios/oVirt_concepts.rst [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+Sorry, but Windows cannot be used as a control node.
+There can be multiple control nodes.
+
+
+Delegation
+==========
+
+If you want to perform a oVirt specific task on one host within a RHV Datacenter,
+use the ``delegate_to`` keyword on a task.
+This delegation host will be any host where you have ``pyVmomi`` installed.
```
Comment (Akasurde) [inline docs/docsite/rst/scenario_guides/ovirt_scenarios/oVirt_concepts.rst:33]: Replace pyVmomi with `ovirt-engine-sdk-python` library.

File: docs/docsite/rst/scenario_guides/ovirt_scenarios/oVirt_concepts.rst [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+The ordered lists of tasks, saved so you can run those tasks in that order
+repeatedly and practice infrastructure-as-code. Playbooks can include variables
+as well as tasks. Playbooks are written in YAML and are easy to read, write,
+share and understand.
+
+
+ovirt-engine-sdk-python
+=======================
+
+Ansible oVirt modules often require the ovirt-engine-sdk-python package installed.
```
Comment (Akasurde) [inline docs/docsite/rst/scenario_guides/ovirt_scenarios/oVirt_concepts.rst:60]:
```suggestion
Ansible oVirt modules often require the ``ovirt-engine-sdk-python`` package installed.
```

File: docs/docsite/rst/scenario_guides/ovirt_scenarios/oVirt_intro.rst [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+    $ sudo yum install ovirt-engine-sdk-python
+
+
+ovirt_* modules
+===================
+
+The modules for oVirt are all known generally as ``ovirt_<something>``
+
+They are found here: `oVirt Modules <https://docs.ansible.com/ansible/latest/modules/list_of_cloud_modules.html#ovirt>`
```
Comment (Akasurde) [inline docs/docsite/rst/scenario_guides/ovirt_scenarios/oVirt_intro.rst:32]:
```suggestion
They are found here: `oVirt Modules <https://docs.ansible.com/ansible/latest/modules/list_of_cloud_modules.html#ovirt>`_
```

File: docs/docsite/rst/scenario_guides/ovirt_scenarios/scenario_remove_vm.rst [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+This guide will show you how to utilize Ansible to remove an existing oVirt virtual machine.
+
+Scenario Requirements
+=====================
+
+* Software
+
+    * Ansible 2.5 or later must be installed.
+
+    * The Python module ``Pyvmomi`` must be installed on the Ansible control node (or Target host if not executing against localhost).
```
Comment (Akasurde) [inline docs/docsite/rst/scenario_guides/ovirt_scenarios/scenario_remove_vm.rst:22]:
```suggestion
    * The Python module ``ovirt-engine-sdk-python`` must be installed on the Ansible control node (or Target host if not executing against localhost).
```

File: docs/docsite/rst/scenario_guides/ovirt_scenarios/scenario_remove_vm.rst [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+
+    * We recommend installing the latest version with pip: ``pip install Pyvmomi`` (as the OS packages are usually out of date and incompatible).
```
Comment (Akasurde) [inline docs/docsite/rst/scenario_guides/ovirt_scenarios/scenario_remove_vm.rst:24]:
```suggestion
    * We recommend installing the latest version with pip: ``pip install ovirt-engine-sdk-python`` (as the OS packages are usually out of date and incompatible).
```

File: docs/docsite/rst/scenario_guides/ovirt_scenarios/scenario_remove_vm.rst [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+          register: facts
+
+
+Since Ansible utilizes the oVirt API to perform actions, in this use case it will be connecting directly to the API from localhost.
+
+This means that playbooks will not be running from the oVirt/RHV engine or host Server.
+
+Note that this play disables the ``gather_facts`` parameter, since you don't want to collect facts about localhost.
+
+You can run these modules against another server that would then connect to the API if localhost does not have access to oVirt/RHV engine. If so, the required Python modules will need to be installed on that target server. We recommend installing the latest version with pip: ``pip install Pyvmomi`` (as the OS packages are usually out of date and incompatible).
```
Comment (Akasurde) [inline docs/docsite/rst/scenario_guides/ovirt_scenarios/scenario_remove_vm.rst:86]:
```suggestion
You can run these modules against another server that would then connect to the API if localhost does not have access to oVirt/RHV engine. If so, the required Python modules will need to be installed on that target server. We recommend installing the latest version with pip: ``pip install ovirt-engine-sdk-python`` (as the OS packages are usually out of date and incompatible).
```

File: docs/docsite/rst/scenario_guides/ovirt_scenarios/scenario_rename_vm.rst [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+            uuid: "{{ vm_facts.instance.hw_product_uuid }}"
+            name: "{{ new_vm_name }}"
+
+Since Ansible utilizes the oVirt API to perform actions, in this use case it will be connecting directly to the API from localhost.
+
+This means that playbooks will not be running from the engine or host Server.
+
+Note that this play disables the ``gather_facts`` parameter, since you don't want to collect facts about localhost.
+
+You can run these modules against another server that would then connect to the API if localhost does not have access to engine. If so, the required Python modules will need to be installed on that target server. We recommend installing the latest version with pip: ``pip install Pyvmomi`` (as the OS packages are usually out of date and incompatible).
```
Comment (Akasurde) [inline docs/docsite/rst/scenario_guides/ovirt_scenarios/scenario_rename_vm.rst:93]:
```suggestion
You can run these modules against another server that would then connect to the API if localhost does not have access to engine. If so, the required Python modules will need to be installed on that target server. We recommend installing the latest version with pip: ``pip install ovirt-engine-sdk-python`` (as the OS packages are usually out of date and incompatible).
```

File: docs/docsite/rst/scenario_guides/ovirt_scenarios/scenario_rename_vm.rst [hunk 10]
```diff
@@ -0,0 +1,170 @@
...
+This value will be used input for ``oVirt_*`` modules. Specify new name to virtual machine which conforms to all oVirt requirements for naming conventions as ``name`` parameter. Also, provide ``uuid`` as the value of oVirt UUID.
+
+What to expect
+--------------
+
+Running this playbook can take some time, depending on your environment and network connectivity. When the run is complete you will see
+
+.. code-block:: yaml
+
+    {
```
Comment (Akasurde) [inline docs/docsite/rst/scenario_guides/ovirt_scenarios/scenario_rename_vm.rst:119]: Replace this with actual output of `ovirt_` module output.

File: docs/docsite/rst/scenario_guides/ovirt_scenarios/oVirt_external_doc_links.rst [hunk 11]
```diff
@@ -0,0 +1,15 @@
+.. _rhv_external_doc_links:
+
+***********************************
+List of useful links for oVirt/ oVirt
```
Comment (Akasurde) [inline docs/docsite/rst/scenario_guides/ovirt_scenarios/oVirt_external_doc_links.rst:4]:
```suggestion
List of useful links for oVirt/RHV
```

File: docs/docsite/rst/scenario_guides/ovirt_scenarios/oVirt_external_doc_links.rst [hunk 12]
```diff
@@ -0,0 +1,15 @@
...
+***********************************
+
+The following are various external documents and guides, on the topic of 
+oVirt/ oVirt and Ansible.
```
Comment (Akasurde) [inline docs/docsite/rst/scenario_guides/ovirt_scenarios/oVirt_external_doc_links.rst:8]:
```suggestion
oVirt/RHV and Ansible.
```

File: docs/docsite/rst/scenario_guides/ovirt_scenarios/faq.rst [hunk 13]
```diff
@@ -0,0 +1,34 @@
+.. _oVirt_faq:
+
+***************
+Ansible oVirt FAQ
+***************
+
+ovirt_* modules
+===============
+
+What is oVirt? What is oVirt?
```
Comment (Akasurde) [inline docs/docsite/rst/scenario_guides/ovirt_scenarios/faq.rst:18]:
```suggestion
What is oVirt? What is RHV?
```

**Review by Akasurde — CHANGES_REQUESTED** [review]

File: docs/docsite/rst/scenario_guides/ovirt_scenarios/faq.rst [hunk 14]
```diff
@@ -0,0 +1,35 @@
...
+maturity of oVirt insists that it is prime for enterprise production workloads, among others.
+
+What version of oVirt/RHV will Ansible modules work with?
+-----------------------------------------------------------
+
+Most of the more recently build ``ovirt_*`` modules are best experienced with oVirt/RHV 4.1 or newer, hence the preview flags 
+on the modules, indicating no guarantee for backwards compatibility.
+
+Can I deploy a virtual machine on a standalone oVirt Host server ?
+----------------------------------------------------------------
```
Comment (Akasurde) [inline docs/docsite/rst/scenario_guides/ovirt_scenarios/faq.rst]:
```suggestion
Can I deploy a virtual machine on a standalone oVirt Host server ?
------------------------------------------------------------------
```

File: docs/docsite/rst/scenario_guides/ovirt_scenarios/faq.rst [hunk 15]
```diff
@@ -0,0 +1,35 @@
...
+===============
+
+What is oVirt? What is RHV?
+---------------------------
+
+oVirt is the upstream open-source edition of Red Hat Virtualization. 
+Red Hat Virtualization is an open-source alternative hypervisor to such like VMware. As of oVirt version 4.1+ the feature-set and 
+maturity of oVirt insists that it is prime for enterprise production workloads, among others.
+
+What version of oVirt/RHV will Ansible modules work with?
```
Comment (Akasurde) [inline docs/docsite/rst/scenario_guides/ovirt_scenarios/faq.rst:31]:
```suggestion
What version of oVirt/RHV will Ansible modules work with?
---------------------------------------------------------
```

File: docs/docsite/rst/scenario_guides/ovirt_scenarios/faq.rst [hunk 16]
```diff
@@ -0,0 +1,35 @@
...
+Most of the more recently build ``ovirt_*`` modules are best experienced with oVirt/RHV 4.1 or newer, hence the preview flags 
+on the modules, indicating no guarantee for backwards compatibility.
+
+Can I deploy a virtual machine on a standalone oVirt Host server ?
+----------------------------------------------------------------
+
+Yes. ``ovirt_vm`` can deploy a virtual machine with required settings on a standalone (local-storage install) oVirt Host server.
+An arrangement of other ``ovirt_*`` modules combined can achieve more complex VM deployments.
+
+Does oVirt/RHV have a REST APT?
```
Comment (Akasurde) [inline docs/docsite/rst/scenario_guides/ovirt_scenarios/faq.rst:47]:
```suggestion
Does oVirt/RHV have a REST APT?
-------------------------------
```

File: docs/docsite/rst/scenario_guides/ovirt_scenarios/oVirt_scenarios.rst [hunk 17]
```diff
@@ -0,0 +1,23 @@
...
+
+To get started, please select one of the following topics.
+
+
+.. toctree::
+   :maxdepth: 2
+
+   scenario_rename_vm
+   scenario_remove_vm
+   scenario_find_vm_folder
```
Comment (Akasurde) [inline docs/docsite/rst/scenario_guides/ovirt_scenarios/oVirt_scenarios.rst:19]: This is non-existing document reference.

**Comment by Akasurde** [comment]
@funixz I updated few things in this PR. You can build docs locally by installing [`sphinx`](http://www.sphinx-doc.org/en/master/) and 

```
cd <Ansible_checkout>/docs/docsite
rst=scenario_guides/guide_ovirt.rst make htmlsingle
```

Check errors and warnings

**Comment by samccann** [comment]
Please remove all mention of RHV  and keep this as oVirt only please, unless someone in  Red Hat is signing up to officially support these modules.

**Comment by funixz** [comment]
> Please remove all mention of RHV and keep this as oVirt only please, unless someone in Red Hat is signing up to officially support these modules.

will do

**Comment by acozine** [comment]
@funixz you can fix the `not-in-toc-tree` errors by either adding the page(s) in question to a TOC, or marking them `:orphan:`. See https://raw.githubusercontent.com/ansible/ansible/devel/docs/docsite/rst/scenario_guides/vmware_scenarios/vmware_getting_started.rst for an example.

**Comment by samccann** [comment]
@funixz can you please fix the merge commit comment above and rebase etc?

I will add some general comments in a review as well.

**Comment by samccann** [comment]
Some general comments:

- include only files/sections that have actual content. If you want to keep the other sections as reminders to yourself or another contributor, you can comment them out so they won't generate HTML.  

- remove the list of modules. By the time we get this in shape and published, they'll all have moved to a collection so you can just point there. For now, they aren't needed.

- remove the basic stuff like what is a playbook, etc. That is already covered in the ansible docs. You can link to https://docs.ansible.com/ansible/latest/user_guide/basic_concepts.html#ansible-concepts for example (though please use an rst :ref: for that.

I'll take a deeper look once you've cleaned these bigger items up.  Thanks!

**Comment by funixz** [comment]
Sorry I haven't been able to jump back onto this. Once I can slide in the time, I will. I haven't forgotten.

**Comment by samccann** [comment]
At this point, scenario guides should be added or moved to the most appropriate collection repository.  See https://github.com/ansible-collections/kubernetes.core/pull/143 and https://github.com/ansible/ansible/pull/75249 for an example of how this is done.

Thanks @arendej for the work you put into this!

**Pull request closed without merging**